libs/libc/pthread/pthread_once.c

```diff
@@ -75,23 +73,21 @@ int pthread_once(FAR pthread_once_t *once_control,
 
   /* Prohibit pre-emption while we test and set the once_control. */
 
-  nxrmutex_lock(&g_lock);
+  pthread_mutex_lock(&once_control->mutex);
 
-  if (!*once_control)
+  if (!once_control->done)
     {
-      *once_control = true;
+      once_control->done = true;
```

Comment:
I think that `done` should be moved after `init_routine` to satisfy POSIX:
```
The pthread_once() function is not a cancellation point. However, if init_routine is a cancellation point and is canceled, the effect on once_control shall be as if pthread_once() was never called.
```

Also the question about calling a cancellation point API is still in place. How do we release a mutex if the cancellation point API is called?

Reply:
it's no difference for normal mutex, since the dead lock will happen if the code try to hold the same mutex reclusively in the same thread.